### DISCUSSION TITLE
What is the difference between a verifiable mock and a non-verifiable mock?

### DISCUSSION THREAD

**Discussion by splatteredbits** [pester/Pester#2257] [Q&A] [answered]
I can't find any explanation in the docs on the difference between a verifiable mock and non-verifiable mock. 

- When would I use `-Verifiable`?
- When would I not use `-Verifiable`?
- What does `Should -InvokeVerifiable` do to verify my mocks?
- What makes a mock `-Verifiable`? 
- Can I create mocks that `Should -InvokeVerifiable` can't verify?

**Comment by fflaten** [comment, marked as answer]
> **Should -InvokeVerifiable​**
> 
> Checks if any verifiable Mock has not been invoked. If so, this will throw an exception

[Source](https://pester.dev/docs/usage/mocking#should--invokeverifiable)

`Should -InvokeVerifiable` will assert that all mocks with `-Verifiable` has been executed at least once. If one or more has not been invoked yet, the assertion fails.

It could save you from writing many `Should -Invoke` and might be useful as a baseline check for critical mocks.

You can still call `Should -Invoke` if you need to be specific for certain mocks like checking additional parameters besides the mandatory ones in the mock's ParameterFilter or if a mock was called multiple times ++.
